### PR TITLE
Refactor game logic

### PR DESCRIPTION
**Pull request by ohagert1**

### CODE REVIEW

File: client/components/FirstVendor.js [hunk 1]
```diff
@@ -36,53 +40,59 @@ export class FirstVendor extends React.Component {
     this.grade = this.grade.bind(this)
   }
 
-  converse(repeating) {
-    if(this.state.promptIndex < this.props.prompts.length) {
-      if(!repeating) {
+  converse() {
+    let currentPrompt = this.props.currentPrompt
+    if(!Object.keys(this.props.currentPrompt).length) {
```
Comment (schai6) [inline client/components/FirstVendor.js]: let's change `this.props.currentPrompt` to `currentPrompt`.

File: client/components/Hint.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import 'aframe'
+import { Entity } from 'aframe-react'
+import 'babel-polyfill'
```
Comment (schai6) [inline client/components/Hint.js:4]: don't need this and aframe-environment-component and aframe I think

File: server/db/models/character.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const db = require('../db')
```
Comment (schai6) [inline server/db/models/character.js:1]: So we are getting rid of the Character model, so this can be deleted.

File: server/db/index.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 const db = require('./db')
 const {
-  Item,
+  Quest,
+  Character,
```
Comment (schai6) [inline server/db/index.js]: remove Character

File: server/db/index.js [hunk 5]
```diff
@@ -1,6 +1,9 @@
...
+  PromptResponse,
+  CharacterPrompt,
```
Comment (schai6) [inline server/db/index.js]: and CharacterPrompt

File: server/db/index.js [hunk 6]
```diff
@@ -12,7 +15,10 @@ require('./models')
 
 module.exports = {
   db,
-  Item,
+  Quest,
+  Character,
```
Comment (schai6) [inline server/db/index.js]: same here

File: server/db/index.js [hunk 7]
```diff
@@ -12,7 +15,10 @@ require('./models')
...
+  PromptResponse,
+  CharacterPrompt,
```
Comment (schai6) [inline server/db/index.js]: and here

File: seed.js [hunk 8]
```diff
@@ -76,7 +133,7 @@ function seed(prompts, responses, languages) {
 db.sync({force: true})
   .then(() => {
     console.log('Seeding database')
-    return seed(prompts, responses, languages)
+    return seed(prompts, responses, promptResponses, languages, characterPrompts, characters)
```
Comment (schai6) [inline seed.js]: don't need characterPrompts or characters anymore

File: server/db/models/characterPrompt.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+const Sequelize = require('sequelize')
```
Comment (schai6) [inline server/db/models/characterPrompt.js]: can delete this file too

File: server/db/models/index.js [hunk 10]
```diff
@@ -7,16 +10,27 @@ const Response = require('./response')
...
-Scene.belongsToMany(Item, { through: 'scene_items' })
-Prompt.hasMany(Response)
+
+Scene.belongsToMany(Quest, { through: 'scene_quests' })
+Quest.belongsToMany(Scene, { through: 'scene_quests' })
+
+Scene.belongsToMany(Character, { through: 'scene_characters' })
+Character.belongsToMany(Scene, { through: 'scene_characters' })
+
+Prompt.belongsToMany(Character, { through: CharacterPrompt })
```
Comment (schai6) [inline server/db/models/index.js]: Don't need this

File: server/db/models/index.js [hunk 11]
```diff
@@ -7,16 +10,27 @@ const Response = require('./response')
...
-Item.belongsToMany(Prompt, { through: 'prompt_items' })
-Prompt.belongsToMany(Item, { through: 'prompt_items' })
-Item.belongsToMany(Scene, { through: 'scene_items' })
-Scene.belongsToMany(Item, { through: 'scene_items' })
-Prompt.hasMany(Response)
+
+Scene.belongsToMany(Quest, { through: 'scene_quests' })
+Quest.belongsToMany(Scene, { through: 'scene_quests' })
+
+Scene.belongsToMany(Character, { through: 'scene_characters' })
```
Comment (schai6) [inline server/db/models/index.js]: Don't need this

File: server/router/api/promptRouter.js [hunk 12]
```diff
@@ -1,8 +1,13 @@
 const router = require('express').Router()
-const { Prompt } = require('../../db')
+const { Prompt, CharacterPrompt } = require('../../db')
 
 router.get('/', (req, res, next) => {
-  Prompt.scope('populated').findAll()
+  CharacterPrompt.findAll()
```
Comment (schai6) [inline server/router/api/promptRouter.js]: should just fetch all prompts now

**Review by schai6 — CHANGES_REQUESTED** [review]
So, this PR includes the model changes, so it has a lot more than it should. I think I'll make the model changes a PR which can get merged in first, and then pulled into this branch so that it will be clearer.

**Review by leighembee — APPROVED** [review]
looks good

**Review by schai6 — APPROVED** [review]

**Pull request merged**